lib/components/MenuItem/MenuItem.tsx

```diff
@@ -36,35 +30,61 @@ const {
   MENU_ITEM_HOVER,
 } = actionTypes;
 
-export const MenuItem = ({ children, onClick, data }: MenuItemProps) => {
+const menuItemChildrenSize = 'standard';
+
+interface UseMenuItemProps {
+  onClick?: () => void;
+  data?: DataAttributeMap;
+  displayName?: string;
+}
+function useMenuItem<MenuItemElement extends HTMLElement>({
+  displayName = 'MenuItem',
+  onClick,
+  data,
+}: UseMenuItemProps) {
   const styles = useStyles(styleRefs);
   const menuContext = useContext(MenuContext);
 
   assert(
     menuContext !== null,
-    'A MenuItem must be rendered as an immediate child of a Menu. See the documentation for correct usage: https://seek-oss.github.io/braid-design-system/components/MenuRenderer',
+    `${displayName} must be rendered as immediate children of a menu. See the documentation for correct usage: https://seek-oss.github.io/braid-design-system/components/MenuItem`,
```

Comment:
The `displayName` needs to be pluralised here to be consistent with the rest of the sentence.